test/benchmark-common-tasks/collect_stats.py

```diff
@@ -0,0 +1,107 @@
+import os
+import sys
+import getopt
+
+
+def get_file_size(filename):
+    statinfo = os.stat(filename)
+    return statinfo.st_size
+
+
+def get_library_size(rootBuildDir):
+    # librealm.a is the same binary name on mac/linux
+    return get_file_size(os.path.join(rootBuildDir, "src/realm/librealm.a"))
+
+
+def get_loc(directory, recursive=True):
+    loc = 0
+    for root, dirs, files in os.walk(directory, followlinks=False):
+        if (recursive is False and root != directory):
+            continue
+        for filename in files:
+            if (filename.endswith('.hpp') or filename.endswith('.cpp')):
+                fullpath = os.path.join(root, filename)
+                with open(fullpath, 'r') as f:
+                    # use strip to exclude empty lines
+                    loc += len(list(filter(lambda x: x.strip(), f)))
+    return loc
+
+
+def get_lines_of_code(rootSourceDir):
+    return get_loc(os.path.join(rootSourceDir, "src"))
+
+
+def get_lines_of_test_code(rootSourceDir):
+    loc = get_loc(os.path.join(rootSourceDir, "test"), recursive=False)
+    loc += get_loc(os.path.join(rootSourceDir, "test/util"), recursive=False)
+    return loc
+
+
+def get_realm_sizes(rootBuildDir):
+    pathPrefix = os.path.join(rootBuildDir, "test/benchmark-common-tasks/")
+    results = []
+    sizes = [0, 1000, 2000, 4000, 8000, 10000]
+    for size in sizes:
+        name = str(size) + "bytes.realm"
+        os.system(pathPrefix + "realm-stats " + name + " " + str(size))
+        results.append((str(size) + " byte blob", get_file_size(name)))
```

Comment:
The question is of course if there is a good correlation between the raw library file size and the footprint of core in the final application.